File: Source/Domain/Entities/ToDoAssistant/Share.cs
using PersonalAssistant.Domain.Entities.Common;

namespace PersonalAssistant.Domain.Entities.ToDoAssistant
{
    public class Share : Entity
    {
        public int ListId { get; set; }
        public int UserId { get; set; }
        public bool IsAdmin { get; set; }
        public bool? IsAccepted { get; set; }
        public short? Order { get; set; }
        public bool NotificationsEnabled { get; set; }
        public bool IsArchived { get; set; }

        public ToDoList List { get; set; }
        public User User { get; set; }
    }
}

File: src/Clients/to-do-assistant2/src/lib/models/viewmodels/share.ts
export class Share {
	constructor(
		public userId: number,
		public email: string,
		public imageUri: string,
		public isAdmin: boolean,
		public isAccepted: boolean,
		public createdDate: string | null
	) {}
}

export class ShareUserAndPermission {
	constructor(public userId: number, public isAdmin: boolean) {}
}
